elastic/kibana
# Improves screen reader features of combo boxes in edit role screen

## Summary

Related to #27749

While the [EuiComboBox rebuild](https://github.com/elastic/eui/issues/2841) is in progress, this PR addresses missing aria properties and fixes option announcements for the `run as user` combo. It additionally adds an aria-label for the Spaces Navigation EuiSelectable that I found was missing.

### Testing
- Add all sample data
- Edit a non-reserved role
- Turn on a screen reader (NVDA is recommended, but VoiceOver will do)
- Tab through the controls on the Edit role screen. When you arrive in a combo box you may or may not hear an announcement declaring the name of the combo box (this is expected, especially with VoiceOver).
- With focus in a combo box press the down arrow key. Verify the options are announced as you traverse them with the down and up arrow keys.
- If there was success in option announcement, press escape key and verify that the name of the combo box is announced. You may have to press escape twice when using NVDA.

Note: It seems to be the consensus that this is best we can do with the existing implementation of EuiComboBox, which is partly why it is being rebuilt with the EuiSelectable component.
